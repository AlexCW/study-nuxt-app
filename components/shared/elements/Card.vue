<template>
  <div class="card">
    <div 
      v-if="text.length > 0" 
      class="card-text">
      {{ text }}
    </div>
  </div>
</template>

<script>
    export default {
      props: {
        text: {
          default: '',
          type: String
        }
      }
    }
</script>

<style>
.card {
  background: url('/vue-logo.png') no-repeat;
  background-position: center; 
  background-size: cover;
  border-radius: 4px;
  box-shadow: 0px 0px 1px 0px darkgray;
  align-items:flex-start;
  color: white;
  margin-left: 20px;
  margin-top: 20px;
  padding: 100px 150px;
  position: relative;
  flex-basis: 15%;
}
.card .card-text {
  position: absolute;
  bottom: 0;
  left: 0;
  padding: 5px 0px 10px 10px;
  font-weight: 500;
  background: black;
  opacity: 0.5; 
  width: 100%;
}
</style>