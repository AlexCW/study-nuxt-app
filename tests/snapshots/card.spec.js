import { shallowMount } from '@vue/test-utils'
import Card from "@/components/shared/elements/Card"

const factory = () => shallowMount(Card, {
    propsData: {
      text: 'Test Card Text'
    }
  })
  
  describe('Card', () => {
    test('mounts properly', () => {
      const wrapper = factory()
      expect(wrapper.isVueInstance()).toBeTruthy()
    })
  
    test('renders properly', () => {
      const wrapper = factory()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })