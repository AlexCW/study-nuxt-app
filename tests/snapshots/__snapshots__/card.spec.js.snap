// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders properly 1`] = `
<div class="card">
  <div class="card-text">
    Test Card Text
  </div>
</div>
`;
